var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -92,7 +92,13 @@ class Proj(CMakePackage, AutotoolsPackage):
         # CMake 3.19 refactored the FindTiff module interface, update older proj's
         # to be compatible with this "new" interface
         # patch replaces the TIFF_LIBRARY variable (no longer used) with TIFF_LIBRARIES
-        patch("proj-8.1-cmake-3.29-new-tiff-interface.patch", when="+tiff @8:9.1.0 ^cmake@3.19:")
+        patch(
+            "proj-8.1-cmake-3.29-new-tiff-interface.patch", when="+tiff @7:9.1.0 ^cmake@3.19:3.27"
```

Comment:
From https://github.com/spack/spack/pull/44367 Proj 7 does not work with this patch, we should update the version constraints here to reflect that (and not revert that merged change)

Reply:
I have created a new patch in https://github.com/spack/spack/pull/44535 to target @7